### ISSUE TITLE
Threading: close a thread

### ISSUE THREAD

**Issue by mrsimpson** [RocketChat/Rocket.Chat#13664]
## Motivation

Usually, a thread is being created to discuss something with a goal. Once this goal has been achieved, the thread is going to be deserted. Semantically, it's "done".

## Proposal

The same semantic already exists for a room: "Archived". Once a room is archived, it can still be read, but subscriptions are hidden (so that the side panel is getting cleaned up) and the room is write-protected. 
The "archiving" function could be re-used, it just needs to be made more prominent. A dedicated button either next to the message area or in the room properties (maybe even replacing leave/hide) would help.
In addition to that, the status "closed" should be visualized on the thread-started-message in the parent room.

**Comment by engelgabriel** [comment]
https://github.com/RocketChat/Rocket.Chat/issues/13843

**Comment by engelgabriel** [comment]
https://github.com/RocketChat/Rocket.Chat/issues/13644

**Comment by ChrissW-R1** [comment]
It would also be nice to have a permission to allow users to leave discussion, but not public channels or private groups!

**Comment by softwaredoug** [comment]
It'd be great also to be able to turn a Discussion into a room or private group

**Comment by charliehelmijr** [comment]
Is this something that will be implemented?

**Comment by itsarectangle** [comment]
Wait a sec, you mean there's no way to leave or delete a Discussion?  I'm fighting with trying to get a Discussion out of the left sidebar, can't find a way to make it go away!  I don't want to re-assign it to a new owner, I want it gone.  How to make it go away?

**Comment by itsarectangle** [comment]
This thread died?  These are great ideas.  Anyone have links to these ideas being addressed in other threads?

**Comment by itsarectangle** [comment]
Also posted here:  https://github.com/RocketChat/Rocket.Chat/issues/13689

**Comment by ntextreme3** [comment]
Being able to "close" a discussion I think would make it a nice choice for PR review integration. Ex: A new PR starts a discussion. Within the review, you might talk about things and there might be some things that thread out, but is part of the same discussion. Ultimately, the discussion get closed when the PR gets closed. 

In https://github.com/integrations/slack/issues/660 the request is for PR comments to go into a slack thread. Seems like rocket chat's "discussion" would be able to provide a better workflow, assuming the discussion eventually ends.

**Comment by thw0rted** [comment]
Could this please get a new milestone/priority?  Maybe some use-case detail will help here.

My use of RC is on a corporate server that has message deletion disabled for record-keeping purposes, so average users can create a message then promote it to a Discussion, but once they have done this they are the "owner" for the discussion and cannot "leave" it (without reassigning ownership to someone else).  When the Discussion has served its purpose, they can "hide" it in the sidebar but cannot prevent someone else from finding it later and adding a reply, bring it back into the sidebar.

In short, on a server where regular users are not permitted to delete messages, there should be a way to designate that a Discussion should not be used anymore and is considered "historic" and inactive, similar to archiving unused channels for preservation.
